source/concepts/block.rst

```diff
@@ -64,19 +64,26 @@ The consensus algorithm determines a new account to harvest the subsequent block
 The harvesting account receives the :doc:`fees <fees>` for the transactions added in the block and the mosaics created by :doc:`inflation <inflation>`.
 This gives the harvester an incentive to add as many transactions to the block as possible.
 
-.. _rollbacks:
+.. _finalization:
 
-*********
-Rollbacks
-*********
+************
+Finalization
+************
 
-Blockchains are designed in a way that, in the presence of a network failure or partition, the recent blocks might need to be rolled back.
+Finalization is the process of making changes on a blockchain ledger permanent.
+Before blocks reach finality, they can be rolled back in the presence of a network failure or partition.
+Once blocks completes finalization, they are immutable.
 
-The rewrite limit is the maximum number of blocks that can be rolled back.
-Hence, forks can only be resolved up to a certain depth too.
+For every block, a sorting algorithm selects zero, one, or multiple accounts to propose the block to be finalized; these selected accounts unveil themselves at the beginning of the iteration to other nodes.
+Then, the committee selection algorithm verifies the eligible accounts to vote on the finality of the proposed block.
```

Comment:
We could describe how to become an eligible account:

1. Own at least [minVoterBalance](https://github.com/nemtech/symbol-openapi/blob/master/spec/core/network/schemas/ChainPropertiesDTO.yml) network currency units.
1. Register as a voter by announcing a [VotingKeyLinkTransaction](https://nemtech.github.io/serialization/coresystem.html#voting-key-link-transaction)